flink-libraries/flink-table/src/main/scala/org/apache/flink/table/functions/utils/AggSqlFunction.scala

```diff
@@ -57,6 +57,8 @@ class AggSqlFunction(
   ) {
 
   def getFunction: AggregateFunction[_, _] = aggregateFunction
+
+  override def isDeterministic: Boolean = aggregateFunction.isDeterministic
```

Comment:
Can we add an abstract class for `AggSqlFunction`, `ScalarSqlFunction`,and `TableSqlFunction`. What do you think? @wuchong @Xpray